circleci/breakdown-sync

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# Parses dependencies and sends info to breakdown via circle-ci-integrations
+#
+# Usage:
+#
+#   breakdown-sync [BREAKDOWN_URL] [USER] [PASS]
+#
+
+set -e
+
+: ${CIRCLE_BRANCH?"Missing required env var"}
+BRANCH=$CIRCLE_BRANCH
+if [[ $BRANCH != "master" ]]; then
+  echo "Skipping sync for non-master branch"
+  exit 0
+fi
+
+DIR=$(dirname "$0")
+. $DIR/utils
+
+# User supplied args
+BREAKDOWN_URL=$1
+if [[ -z $BREAKDOWN_URL ]]; then echo "Missing arg1 BREAKDOWN_URL" && exit 1; fi
+USER=$2
+if [[ -z $USER ]]; then echo "Missing arg2 USER" && exit 1; fi
+PASS=$3
+if [[ -z $PASS ]]; then echo "Missing arg3 PASS" && exit 1; fi
+
+: ${CIRCLE_PROJECT_REPONAME?"Missing required env var"}
+REPO=$CIRCLE_PROJECT_REPONAME
+: ${CIRCLE_SHA1?"Missing required env var"}
+FULL_SHA=${CIRCLE_SHA1}
+SHORT_SHA=${CIRCLE_SHA1:0:8}
+
+# check if repo commit already exists
+RC=$(curl -u $USER:$PASS \
```

Comment:
Not blocking, for this PR but something to consider for the future. Another solution for this would be to have the service itself run this check when submitting a new commit and just silently swallow the request if it already exists.

Reply:
I want to avoid downloading and running the CLI as it'll take a quite a few seconds. 

But I might go ahead and implement this to avoid hitting alarms in circle-ci-integrations.

Reply:
Long term the CLI and build app will be combined which will reduce the concern of download times. Another options may be to have that endpoint return a 404 instead of a 500 when a commit isn't found which I think will not trigger alarms.

Reply:
Hm, it's already returning 404s. Let me double check that it won't actually trigger the alarms.

Reply:
Ah in that case it may have been other errors this morning that were causing the large amount of errors in the internal error alarm.

Reply:
To be fair, I was unintentionally making the service return a bunch of 500s while I was testing this script. Checking Datadog, the 404s don't register as errors.

The CLI can take up to 25 seconds in CI since there's a fresh go mod cache and it needs that info to parse the dep graph.